spec/requests/spree/api/v2/storefront/vendors_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'spec_helper'
+
+describe 'API V2 Storefront Vendor Spec', type: :request do
+
+  let!(:vendor) { create(:active_vendor, name: 'vendor') }
+  let!(:product) { create(:product, vendor: vendor) }
+
+  describe 'vendors#show' do
+    context 'with invalid slug param' do
+      before { get '/api/v2/storefront/vendors/vendor1' }
+
+      it_behaves_like 'returns 404 HTTP status'
+    end
+
+    context 'with valid slug param' do
+      before { get "/api/v2/storefront/vendors/#{vendor.slug}" }
+
+      it_behaves_like 'returns 200 HTTP status'
+
+      it 'returns one vendor' do
+        json_response = JSON.parse(response.body)
+        expect(json_response.count).to eq 1
+        expect(json_response['data']['type']).to eq('vendor')
+      end
+
+      it 'does not return included' do
+        json_response = JSON.parse(response.body)
+        expect(json_response.keys).to contain_exactly('data')
+      end
+    end
+
+    context 'with products and images included' do
+      before { get "/api/v2/storefront/vendors/#{vendor.slug}?include=products,image" }
+
+      it_behaves_like 'returns 200 HTTP status'
+
+      it 'returns product information' do
```

Comment:
@poudelprakash let's also add a check for image attributes, thanks!

Reply:
I have now added a check for image attributes as well